books.php

```diff
@@ -1,46 +1,36 @@
 <?php
-class books
-{	
-private static $instance = null;
-private function __construct()
+class Books
 {
-$this->link = new PDO('mysql: host=localhost;dbname=library_bd', "root","root");
-}
-private function __clone() {}
-public static function getInstance()
-{
-        if (self::$instance === null)
-        {
-            self::$instance = new books;
+    private static $instance = null;
+    private function __construct()
+    {
+        $this->link = new PDO('mysql: host=localhost;dbname=library_bd', "root", "root");
+    }
+    private function __clone()
+    { }
+    public static function getInstance()
+    {
+        if (self::$instance === null) {
+            self::$instance = new Books;
         }
         return self::$instance;
+    }
+    public function allBooks()
+    {
+        $query = "SELECT * FROM book";
+        $resQuery = $this->link->query($query);
+        echo (json_encode($resQuery->fetchAll(PDO::FETCH_ASSOC)));
+    }
+    public function booksId($id)
+    {
+        $query = "SELECT * FROM book WHERE id = $id";
+        $resQuery = $this->link->query($query);
+        echo json_encode($resQuery->fetch(PDO::FETCH_ASSOC));

```

Comment:
You can use fetchColumn to obtain one value.
https://www.php.net/manual/en/pdostatement.fetchcolumn.php

Reply:
fetchColumn gives only one first value of row, but I want all values of needed row (description chosen books by id)